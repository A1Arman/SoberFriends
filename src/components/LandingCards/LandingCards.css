.card {
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    max-width: 40rem;
    border-radius: 0.5rem;
    margin: 0 3rem 2rem 3rem;
    background: rgba(191, 122, 25, 0.37);
}

img {
    width: 7.5rem;
    height: auto;
}

.card:hover {
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}


@media screen and (min-width: 600px) {
    .card-container {
        display: grid;
        align-items: center;
        justify-content: center;
    }
}

@media screen and (min-width: 992px) {
    .card-container {
        display: grid;
        grid-template-areas: ". title ."
                            "card1 card2 card3";
        align-items: center;
        justify-content: center;
    }

    img {
        grid-area: img;
        margin: 0 auto;
    }

    .card-content {
        text-align: initial;
        grid-area: content;
        padding-left: 0.5rem;
    }

    .card-title {
        margin-block-end: 0.75em;
        grid-area: title;
    }

    .landing-card {
        display: grid;
        grid-template-areas: "img"
                             "title"
                             "content";
        justify-content: center;                      
    }

    #card-container-title {
        grid-area: title;
    }

    #card1 {
        height: 365px;
        width: 275px;
        grid-area: card1;
        margin: 0 auto;
    }

    #card2 {
        height: 365px;
        width: 275px;
        grid-area: card2;
        margin: 0 auto;
    }

    #card3 {
        height: 365px;
        width: 275px;
        grid-area: card3;
        margin: 0 auto;
    }
}

