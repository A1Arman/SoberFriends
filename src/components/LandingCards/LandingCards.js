import React from 'react';
import rating from '../../images/rating.svg';
import impact from '../../images/impact.svg';
import clock from '../../images/clock.svg';
import './LandingCards.css';

function LandingCards() {
    return (
        <section className='card-container'>
            <h3>A Better Way To Quitting Your Addiction</h3>
            <section className='card'>
                <img src={rating} />
                <h4>Easy to Use</h4>
                <p>Track your progress and journey with ease</p>
            </section>
            <section className='card'>
                <img src={impact} />
                <h4>Impact Others</h4>
                <p>Connect and meet people going through the same challenges. Keep each other motivated!</p>
            </section>
            <section className='card'>
                <img src={clock} />
                <h4>Create a Lasting Change</h4>
                <p>Create a lasting change by taking daily action while being part of a motivated community.</p>
            </section>
        </section>
    )
}

export default LandingCards;