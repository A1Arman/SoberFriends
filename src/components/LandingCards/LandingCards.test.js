import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LandingCards from '../LandingCards/LandingCards';


describe(`LandingCards component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<LandingCards />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});