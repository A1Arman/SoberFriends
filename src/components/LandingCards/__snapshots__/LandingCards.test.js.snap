// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingCards component renders without crashing 1`] = `
<section
  className="card-container"
>
  <h3
    id="card-container-title"
  >
    A Better Way To Quitting Your Addiction
  </h3>
  <section
    className="card landing-card"
    id="card1"
  >
    <img
      alt="Pointer Finger Icon"
      src="rating.svg"
    />
    <h4
      className="card-title"
    >
      Easy to Use
    </h4>
    <p
      className="card-content"
    >
      Track your progress and journey with ease
    </p>
  </section>
  <section
    className="card landing-card"
    id="card2"
  >
    <img
      alt="People Icon"
      src="impact.svg"
    />
    <h4
      className="card-title"
    >
      Impact Others
    </h4>
    <p
      className="card-content"
    >
      Connect and meet people going through the same challenges. Keep each other motivated!
    </p>
  </section>
  <section
    className="card landing-card"
    id="card3"
  >
    <img
      alt="Clock Icon"
      src="clock.svg"
    />
    <h4
      className="card-title"
    >
      Create a Lasting Change
    </h4>
    <p
      className="card-content"
    >
      Create a lasting change by taking daily action while being part of a motivated community.
    </p>
  </section>
</section>
`;
